### PR TITLE
Implement object operations on SDK client

### PR DESCRIPTION
**Pull request by cthulhu-rider**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,11 +3,13 @@ module github.com/nspcc-dev/neofs-api-go
 go 1.14
 
 require (
+	github.com/davecgh/go-spew v1.1.1
```
Comment (alexvanin) [inline go.mod:6]: No spew, please

File: pkg/client/object.go [hunk 2]
```diff
@@ -0,0 +1,881 @@
...
+
+func (p *ObjectHeaderParams) WithMainFields() *ObjectHeaderParams {
+	if p != nil {
+		p.short = true
+	}
+
+	return p
+}
+
+func (c *Client) GetObjectHeader(ctx context.Context, p *ObjectHeaderParams, opts ...CallOption) (*object.Object, error) {
```
Comment (alexvanin) [inline pkg/client/object.go:485]: Option for short / full header will be set at `*ObjectHeaderPararms`?

File: pkg/client/object.go [hunk 3]
```diff
@@ -0,0 +1,881 @@
...
+	case 2:
+		return c.putObjectV2(ctx, p, opts...)
+	default:
+		return nil, unsupportedProtocolErr
+	}
+}
+
+func (c *Client) putObjectV2(ctx context.Context, p *PutObjectParams, opts ...CallOption) (*object.ID, error) {
+	// convert object to V2
+	obj, err := p.obj.ToV2(c.key)
```
Comment (alexvanin) [inline pkg/client/object.go:178]: Does `p.obj` have set up all available fields at this stage?

File: go.mod (hunk 1 above)
Comment (cthulhu-rider) [reply go.mod]: Fixed

File: pkg/client/object.go (hunk 2 above)
Comment (cthulhu-rider) [reply pkg/client/object.go:485]: Yes, through `WithAllFields`, `WithMainFields` methods.

File: pkg/client/object.go (hunk 3 above)
Comment (cthulhu-rider) [reply pkg/client/object.go:178]: `ID` and `Signature` will be set by `ToV2` method if needed.

Comment (alexvanin) [reply pkg/client/object.go:178]: So, `ToV2` _completes_ object structure?

Comment (cthulhu-rider) [reply pkg/client/object.go:178]: Yes, it completes structure once. If passed `Object` is already completed, `ToV2` does not change it.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**